index.js

```diff
@@ -1,171 +1,126 @@
-'use strict';
-
-module.exports = TinySDF;
-module.exports.default = TinySDF;
-
-var INF = 1e20;
-
-function TinySDF(fontSize, buffer, radius, cutoff, fontFamily, fontWeight) {
-    this.fontSize = fontSize || 24;
-    this.buffer = buffer === undefined ? 3 : buffer;
-    this.cutoff = cutoff || 0.25;
-    this.fontFamily = fontFamily || 'sans-serif';
-    this.fontWeight = fontWeight || 'normal';
-    this.radius = radius || 8;
-
-    // For backwards compatibility, we honor the implicit contract that the
-    // size of the returned bitmap will be fontSize + buffer * 2
-    var size = this.size = this.fontSize + this.buffer * 2;
-    // Glyphs may be slightly larger than their fontSize. The canvas already
-    // has buffer space, but create extra buffer space in the output grid for the
-    // "halo" to extend into (if metric extraction is enabled)
-    var gridSize = size + this.buffer * 2;
-
-    this.canvas = document.createElement('canvas');
-    this.canvas.width = this.canvas.height = size;
-
-    this.ctx = this.canvas.getContext('2d');
-    this.ctx.font = this.fontWeight + ' ' + this.fontSize + 'px ' + this.fontFamily;
+const INF = 1e20;
+
+export default class TinySDF {
+    constructor({
+        fontSize = 24,
+        buffer = 3,
+        radius = 8,
+        cutoff = 0.25,
+        fontFamily = 'sans-serif',
+        fontWeight = 'normal'
+    }) {
+        this.buffer = buffer;
+        this.cutoff = cutoff;
+        this.radius = radius;
+
+        // For backwards compatibility, we honor the implicit contract that the
```

Comment:
Since we don't have the backwards compatibility constraints here, I think we can simplify things a little by just making the canvas bigger:

- Don't need to track a separate `gridSize`
- Don't need to have clipping logic in `getMetrics` (it might still be necessary to handle as an edge case, but it might be simpler to just start with a larger canvas and in the unlikely case that you'd overflow, just resize the canvas to be bigger).